webapp/components/post_view/components/new_message_indicator.jsx

```diff
@@ -0,0 +1,71 @@
+// Copyright (c) 2016 Mattermost, Inc. All Rights Reserved.
+// See License.txt for license information.
+import React from 'react';
+import {FormattedMessage} from 'react-intl';
+
+export default class NewMessageIndicator extends React.Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            visible: false,
+            rendered: false
+        };
+    }
+    componentDidMount() {
+        this.onTransition = this.setRendered.bind(this);
+        this.refs.indicator.addEventListener('transitionend', this.onTransition);
+    }
+    componentWillUnmount() {
+        this.refs.indicator.removeEventListener('transitionend', this.onTransition);
+    }
+    componentWillReceiveProps(nextProps) {
+        if (nextProps.newMessages > 0) {
+            this.setState({rendered: true}, () => {
+                this.setState({visible: true});
+            });
+        } else {
+            this.setState({visible: false});
+        }
+    }
+    render() {
+        let className = 'nav-pills__unread-indicator-bottom';
+        if (this.state.visible > 0) {
```

Comment:
I might be missing something, but I'm a bit confused as to why there's extra state in this component instead of just:
1. When `this.props.newMessages === 0`, it's position is `bottom: -30px;`
2. When `this.props.newMessages > 0`, it's position is `bottom: 0px;`

The extra step to add `display: none` doesn't seem needed when it's already hidden off screen.